vulcanai/models/metrics.py

```diff
@@ -793,9 +794,26 @@ def bootfold_p_estimate(network, data_loader, n_samples, k, epochs,
             p_value : float
         """
         ls_imprv_scores = []
+
+        rand_sample = data.RandomSampler(data_loader.dataset, replacement=True)
+        data_loader_args = inspect.signature(data.DataLoader.__init__)
+        new_params = {}
+
+        for param in data_loader_args.parameters:
```

Comment:
@nirtiac would there be a better way to index the parameters so that you don't need to loop through it? it's not a long signature so it won't take long but it would make it a lot cleaner I think

Reply:
I could do something  like this @rfratila, it looks cleaner but is harder to understand imo. I need to access the signature and then actually get the value of attribute. 

```        new_params = {k:getattr(data_loader, k) for k in data_loader_args.parameters}
        del new_params["self"]
        del new_params["shuffle"]
        new_params["sampler"] = rand_sample
        new_params["batch_sampler"] = None```

Reply:
I guess we are still looping through it so we won't get that much of a speed up. And it's not like the signature will grow longer and longer that we would notice that much of a slow down so I think we should prioritize readability or optimization here. thoughts?

Reply:
yeah that's my opinion too @rfratila

Reply:
Ok let’s revert

Reply:
never changed :P but yes :)